store/tikv/mock-tikv/mvcc_leveldb.go

```diff
@@ -0,0 +1,703 @@
+	startKey := mvccEncode(key, lockVer)
+	iter := mvcc.db.NewIterator(&util.Range{
+		Start: startKey,
+	}, nil)
+	defer iter.Release()
+
+	more := iter.Next()
+	if !more {
+		return nil, nil
+	}
+	return getValue(iter, key, startTS, isoLevel)
+}
+
+func getValue(iter iterator.Iterator, key []byte, startTS uint64, isoLevel kvrpcpb.IsolationLevel) ([]byte, error) {
+	more := true
+	lock, ok, err := meetLock(iter, key)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if ok {
+		if isoLevel == kvrpcpb.IsolationLevel_SI {
+			if lock.startTS <= startTS {
+				return nil, lock.lockErr(key)
+			}
+		}
+		more = iter.Next()
+	}
+
+	for more {
+		value, ok, err := getMvccValue(iter, key)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if !ok {
+			return nil, nil
+		}
+		if value.valueType == typeRollback {
+			more = iter.Next()
+			continue
+		}
+
+		if value.commitTS <= startTS {
+			if value.valueType == typeDelete {
+				return nil, nil
+			}
+			return value.value, nil
+		}
+		more = iter.Next()
+	}
+	return nil, nil
+}
+
+// BatchGet implements the MVCCStore interface.
+func (mvcc *MVCCLevelDB) BatchGet(ks [][]byte, startTS uint64, isoLevel kvrpcpb.IsolationLevel) []Pair {
+	var pairs []Pair
+	for _, k := range ks {
+		v, err := mvcc.Get(k, startTS, isoLevel)
+		if v == nil && err == nil {
+			continue
+		}
+		pairs = append(pairs, Pair{
+			Key:   k,
+			Value: v,
+			Err:   err,
+		})
+	}
+	return pairs
+}
+
+// Scan implements the MVCCStore interface.
+func (mvcc *MVCCLevelDB) Scan(startKey, endKey []byte, limit int, startTS uint64, isoLevel kvrpcpb.IsolationLevel) []Pair {
+	var start, end []byte
+	if len(startKey) > 0 {
+		start = mvccEncode(startKey, lockVer)
+	}
+	if len(endKey) > 0 {
+		end = mvccEncode(endKey, lockVer)
+	}
+	iter := mvcc.db.NewIterator(&util.Range{
+		Start: start,
+		Limit: end,
+	}, nil)
+	defer iter.Release()
+
+	var pairs []Pair
+	more := iter.Next()
+	if more && len(startKey) == 0 {
+		first, _, err := mvccDecode(iter.Key())
+		if err != nil {
+			log.Error("data corrupt:", err)
+			return nil
+		}
+		startKey = first
+	}
+	currKey := startKey
+	for more && len(pairs) < limit {
+		value, err := getValue(iter, currKey, startTS, isoLevel)
+		if err != nil {
+			pairs = append(pairs, Pair{
+				Key: currKey,
+				Err: errors.Trace(err),
+			})
+		}
+		if value != nil {
+			pairs = append(pairs, Pair{
+				Key:   currKey,
+				Value: value,
+			})
+		}
+
+		currKey, more, err = seekToNext(iter, currKey)
+		if err != nil {
+			log.Error("seekToNext error:", errors.ErrorStack(err))
+			break
+		}
+	}
+	return pairs
+}
+
+func seekToNext(iter iterator.Iterator, curr []byte) ([]byte, bool, error) {
+	more := true
+	for more {
+		key, _, err := mvccDecode(iter.Key())
+		if err != nil {
+			return nil, false, errors.Trace(err)
+		}
+		if !bytes.Equal(key, curr) {
+			return key, true, nil
+		}
+		more = iter.Next()
+	}
```

Comment:
how about
```
for {
  // ...
  if !iter.next() {
    break
  }
}
```
instead of
```
more := true
for more {
    // ...
    more = iter.next()
}
```
?